### PR TITLE
docs: FOUNDENG-321 On aizn-admin with PodMan cancelling causes slurm & podman problems

### PR DESCRIPTION
**Pull request by rcorujo**
## Description

Unlike Singularity, Podman starts up several processes when running a container, such as one or more "podman" processes, "catatonit" (responsible for cleaning up container), and "conmon" (runs the processes inside the container).

When a Podman container is running under Slurm and the Slurm job is cancelled, we may end up with a situation where the container remains running, even though Slurm has cancelled the job.  When this occurs on Slurm 21, "podman container ls" will issue an error indicating that "podman system migrate" needs to be run and "sinfo" will show the node in a "drain" state.  When this occurs on Slurm 22, "podman container ls" will show the container still running, although "sinfo" will show the node in an "idle" state.

To handle this situation, we provide instructions in the Podman documentation to create a Slurm task epilog script that will stop the container.

image

image

## Test Plan

Ran a notebook and an experiment and was able to cancel the job in both cases without any podman containers remaining running.

[corujor@znode54 cifar10_pytorch]$ det notebook start --config-file myconfig.yaml

[corujor@znode54 cifar10_pytorch]$ det experiment create rig_const.yaml .

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

**Review by phillip-gaisford — APPROVED** [review]

**Review by jerryharrow — COMMENTED** [review]
I'd prefer this in the Podmand Known Issues section instead of the requirements section.

**Comment by rcorujo** [comment]
> I'd prefer this in the Podmand Known Issues section instead of the requirements section.

Ack

**Review by jerryharrow — APPROVED** [review]
Looks good.   My only comment is that it could use a sentence or two that more clearly defines the problem up-front so people know if it is relevant:

- Compute nodes go into drain state when cancelling a job

- Running a job gets the error level=error msg="invalid internal status, try resetting the pause process with \"/usr/local/bin/podman system migrate\": could not find any running process: no such process"

**Review by jerryharrow — APPROVED** [review]

**Review by jerryharrow — APPROVED** [review]

File: docs/cluster-setup-guide/deploy-cluster/sysadmin-deploy-on-slurm/slurm-known-issues.rst [hunk 1]
```diff
@@ -158,6 +158,50 @@ Some constraints are due to differences in behavior between Docker and Singulari
...
+            slurm_job_name_suffix=$(echo ${SLURM_JOB_NAME} | sed 's/^\S\+-\([a-z0-9]\+-[a-z0-9]\+\)$/\1/')
+
+            podman_container_stop_command="podman container stop \
+               --filter name='.+-${slurm_job_name_suffix}'"
+
+            echo "$(date):$0: Running \"${podman_container_stop_command}\"" 1>&2
+
+            eval ${podman_container_stop_command}
+
+      Restart the ``slurmd`` deamon on all compute nodes.
```
Comment (jerryharrow) [inline docs/cluster-setup-guide/deploy-cluster/sysadmin-deploy-on-slurm/slurm-known-issues.rst:203]: Isn't the more common spelling:  daemon

Comment (rcorujo) [reply docs/cluster-setup-guide/deploy-cluster/sysadmin-deploy-on-slurm/slurm-known-issues.rst:203]: Ah, yes. I'll fix it.

**Review by tara-det-ai — APPROVED** [review]
i made some minor edits for readability

**Comment by rcorujo** [comment]
> i made some minor edits for readability

Thank you.  I see that it is reporting a lint failure.

```
#!/bin/bash -eo pipefail
make -C docs check
make: Entering directory '/home/circleci/project/docs'
git ls-files -z '*.rst' | xargs -0 rstfmt -w 100 --check
cluster-setup-guide/deploy-cluster/sysadmin-deploy-on-slurm/slurm-known-issues.rst is not correctly formatted!
make: *** [Makefile:84: check] Error 123
make: Leaving directory '/home/circleci/project/docs'

Exited with code exit status 2
```

When I run "make -C docs check" in my sandbox, I don't get any errors.

**Review by jerryharrow — APPROVED** [review]

**Comment by jerryharrow** [comment]
| When I run "make -C docs check" in my sandbox, I don't get any errors.

You need to do 'make -C docs fmt'

**Comment by stoksc** [comment]
@cla-bot check

**Comment by phillip-gaisford** [comment]
> This keeps failing, although it has nothing to do with my change. Any idea how to fix this?

Rebase (or start over)?

**Comment by rcorujo** [comment]
> > This keeps failing, although it has nothing to do with my change. Any idea how to fix this?
> 
> Rebase (or start over)?

Did a fresh "git clone", created a new branch, and a new Pull Request.  The failure is still occurring.

https://github.com/determined-ai/determined/pull/5725

**Comment by stoksc** [comment]
let me try to fix it real fast

**Comment by rcorujo** [comment]
Closing this pull request, as the e2e_tests keep failing and it's not related to my doc change. 

Jerry issued a new pull request https://github.com/determined-ai/determined/pull/5760 which was successful.

**Pull request closed without merging**